sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFileFormat.scala

```diff
@@ -202,145 +203,156 @@ class ParquetFileFormat
     (file: PartitionedFile) => {
       assert(file.partitionValues.numFields == partitionSchema.size)
 
-      val filePath = file.toPath
-      val split = new FileSplit(filePath, file.start, file.length, Array.empty[String])
-
+      val split = new FileSplit(file.toPath, file.start, file.length, Array.empty[String])
       val sharedConf = broadcastedHadoopConf.value.value
 
-      val fileFooter = if (enableVectorizedReader) {
```

Comment:
This `buildReaderWithPartitionValues` method is super long now, can we create some smaller methods to split it, so that this PR is easier to review?

Reply:
3 code blocks were extracted from this method as independent methods.